src/arviz_stats/psense.py

```diff
@@ -0,0 +1,193 @@
+"""Power-scaling sensitivity diagnostics."""
+
+from typing import cast
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from arviz_base import extract
+from arviz_base.labels import BaseLabeller
+from arviz_base.sel_utils import xarray_var_iter
+
+from arviz_stats.utils import get_log_likelihood_dataset, get_log_prior
+from arviz_stats.validate import validate_dims
+
+labeller = BaseLabeller()
+
+__all__ = ["psense", "psense_summary"]
+
+
+def psense(
+    dt,
+    group="log_prior",
+    sample_dims=None,
+    component_var_names=None,
+    component_coords=None,
+    var_names=None,
+    coords=None,
+    filter_vars=None,
+    delta=0.01,
+):
+    """
+    Compute power-scaling sensitivity values.
+
+    Parameters
+    ----------
+    data : obj
+        Any object that can be converted to an :class:`arviz.InferenceData` object.
+        Refer to documentation of :func:`arviz.convert_to_dataset` for details.
+        For ndarray: shape = (chain, draw).
+        For n-dimensional ndarray transform first to dataset with ``az.convert_to_dataset``.
+    group : {"log_prior", "log_likelihood"}, default "log_prior"
+        When `component` is "likelihood", the log likelihood values are retrieved
+        from the ``log_likelihood`` group as pointwise log likelihood and added
+        together. With "prior", the log prior values are retrieved from the
+        ``log_prior`` group.
```

Comment:
Do we want `group` argument or `component` argument like in current ArviZ here? The docstring still mentions `component` argument but it is not used anymore.

Reply:
`group` sounds more consistent with other functions. But not opposed to have `component`.

Reply:
From the component api I like it is `prior`/`likelihood` without the log, a bit more conceptual. I think in general `group` is expected to potentially be any of the available groups everywhere except in `plot_ppc` where also similarly we use `posterior`/`prior` but use `posterior_predictive`/`prior_predictive` groups, so using prior/likelihood would align on this end here too. Between `component` or `group` as the argument name I don't really care, but I would update the `component_var_names` and coords if we use `group` (or at least document very clearly), as `group` will probably be more intuitive to users.